src/scraping/collector.py

```
@@ -0,0 +1,123 @@
+    def _build_params(self, criteria: Dict[str, Any], page: int) -> Dict[str, Any]:
+        """Build search parameters"""
+        return {
+            "location": criteria.get("location", ""),
+            "price_min": criteria.get("price_min", 0),
+            "price_max": criteria.get("price_max", 9999999),
+            "property_type": criteria.get("property_type", "any"),
+            "page": page,
+            "limit": min(50, self.batch_size)
+        }
+
+    def _process_listings(self, listings: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
```

Ruff (0.8.2)

Use `list` instead of `List` for type annotation

Replace with `list`

(UP006)